Readme.txt

```diff
@@ -1 +1,80 @@
-hi readme
\ No newline at end of file
+TCP Client-Server Program HW1: Protocol Description
+===============================================
+
+Overview
+--------
+This application implements a TCP client-server system where the server provides multiple 
+calculation services to connected clients. Communication between the server and clients follows 
+a custom application-layer protocol.
+The server supports multiple clients simultaneously using `select` to manage socket connections.
+
+Protocol Description
+---------------------
+
+General Workflow
+----------------
+1. **Server Initialization**:
+   - The server starts with a user file and an optional port parameter.
+   - Users must authenticate with valid credentials stored in the user file.
+
+2. **Client Initialization**:
+   - Clients connect to the server with the command ./numbers_client.py [hostname[port]] and authenticate using a username and password.
+
+3. **Message Format**:
+   - Commands sent by the client and responses sent by the server are terminated by a tab 
+     character (`\t`).
+   - Commands adhere to a strict syntax. Incorrectly formatted messages result in disconnection.
+
+4. **Session Management**:
+   - Clients must authenticate successfully to issue commands.
+   - A `quit` command disconnects the client from the server.
+
+Supported Commands
+------------------
+1. Login:
+   - Client Sends:
+     User: <username>\n<password>\t
+   - Server Responds:
+     - On success: Hi <username>, good to see you.\t
+     - On failure: Failed to login.\t
```

Comment:
Now that i think of it the server can just send 1 for ok and 0 for not connecting, and the client wraps it with the message.
Let me know what you think, its not really an issue...